src/Transform.cpp

```diff
@@ -141,14 +142,29 @@ void Transform::compute_model_matrix(glm::mat4 const& parent_global_model_matrix
     m_parent_dirty = false;
 }
 
+void Transform::set_model_matrix(glm::mat4 const& matrix)
+{
+    m_model_matrix = matrix;
+    if (parent.expired())
+    {
+        glm::decompose(m_model_matrix, m_local_scale, m_local_rotation, m_local_position, m_skew, m_perpective);
+        m_euler_angles = glm::degrees(glm::eulerAngles(m_local_rotation));
+    }
+    else
+    {
+        glm::decompose(glm::inverse(parent.lock()->get_model_matrix()) * m_model_matrix, m_local_scale, m_local_rotation, m_local_position, m_skew, m_perpective);
+        m_euler_angles = glm::degrees(glm::eulerAngles(m_rotation));
+    }
+    set_dirty();
+}
+
 void Transform::compute_local_model_matrix()
 {
     glm::mat4 const transform_x = glm::rotate(glm::mat4(1.0f), glm::radians(m_euler_angles.x), glm::vec3(1.0f, 0.0f, 0.0f));
     glm::mat4 const transform_y = glm::rotate(glm::mat4(1.0f), glm::radians(m_euler_angles.y), glm::vec3(0.0f, 1.0f, 0.0f));
     glm::mat4 const transform_z = glm::rotate(glm::mat4(1.0f), glm::radians(m_euler_angles.z), glm::vec3(0.0f, 0.0f, 1.0f));
 
-    glm::mat4 const rotation_matrix = transform_y * transform_x * transform_z;
-
+    glm::mat4 const rotation_matrix = glm::mat4_cast(m_local_rotation);
```

Comment:
This change and introduction of `m_local_rotation` can be a separate commit as well. Also, lines 163-165 are now unnecessary.